napari/layers/_labels_layer/model.py

```diff
@@ -213,28 +530,147 @@ def get_value(self, position, indices):
         Returns
         ----------
         coord : sequence of int
-            Position of mouse cursor in data.
+            Position of mouse cursor in image coordinates.
         label : int
             Value of the label image at the coord.
-        msg : string
-            String containing a message that can be used as
-            a status update.
         """
         transform = self._node.canvas.scene.node_transform(self._node)
         pos = transform.map(position)
-        pos = [clip(pos[1], 0, self.shape[0]-1), clip(pos[0], 0,
-                                                      self.shape[1]-1)]
+        pos = [np.clip(pos[1], 0, self.shape[0]), np.clip(pos[0], 0,
+                                                          self.shape[1])]
         coord = copy(indices)
-        coord[0] = int(pos[0])
-        coord[1] = int(pos[1])
-        label = self._slice_image(coord, image=self._raw_image)
-        msg = f'{coord}, {self.name}, label {label}'
-        return coord, label, msg
+        coord[0] = pos[0]
+        coord[1] = pos[1]
+        int_coord = copy(coord)
+        int_coord[0] = int(round(coord[0]))
+        int_coord[1] = int(round(coord[1]))
+        label, slice_indices = self._slice_image(int_coord,
+                                                 image=self._raw_image)
+        return coord, label
+
+    def get_message(self, coord, label):
+        """Generates a string based on the coordinates and information about
+        what shapes are hovered over
+
+        Parameters
+        ----------
+        coord : sequence of int
+            Position of mouse cursor in image coordinates.
+        label : int
+            Value of the label image at the coord.
+
+        Returns
+        ----------
+        msg : string
+            String containing a message that can be used as a status update.
+        """
+        int_coord = copy(coord)
+        int_coord[0] = int(round(coord[0]))
+        int_coord[1] = int(round(coord[1]))
+        msg = f'{int_coord}, {self.name}, label {label}'
+
+        return msg
+
+    def on_mouse_press(self, event):
```

Comment:
Now, as I understand it, the model should know nothing about mouse presses or buttons or any such thing, but perhaps this is not the time to start obeying the [commandments](https://github.com/napari/napari/pull/171#issue-259741100).

Reply:
no not ready to refactor the layers yet - but it will come! then we can do them all together